code/game/objects/systems/cell_slot.dm

```diff
@@ -101,3 +134,83 @@
 	obj_cell_slot.remove_yank_time = 0
 	obj_cell_slot.legacy_use_device_cells = TRUE
 	return obj_cell_slot
+
+//? Wrappers for cell.dm functions
+//+ These exist to sanely perform cell functions through the obj_cell_slot system. These break safely if no cell exists.
+
+/**
+ * cell function wrapper - returns the rating of the cell inside or null
+ */
+/datum/object_system/cell_slot/proc/get_rating()
+	return cell?.get_rating()
+
+/**
+ * cell function wrapper - returns the cell src or null
+ */
+/datum/object_system/cell_slot/proc/get_cell(inducer)
+	return cell?.get_cell(inducer)
+
+/**
+ * cell function passthrough - consumes energy using *universal units*
+ * returns amount used or null
+ */
+/datum/object_system/cell_slot/proc/drain_energy(datum/actor, amount, flags)
+	return cell?.drain_energy(actor, amount, flags)
+
```

Comment:
stuff like this should have the @params section of drain_energy

this way dmdoc still functions